application/src/main/java/net/corda/v5/application/persistence/PagedQuery.java

```diff
@@ -49,5 +49,33 @@
      */
     @Suspendable
     @NotNull
-    List<R> execute();
+    ResultSet<R> execute();
+
+    /**
+     * A representation of a result set returned by a paged query when executed.
+     *
+     * @param <T> Type of the results contained in the result set.
+     */
+    interface ResultSet<T> {
+
+        /**
+         * @return The results contained in the result set.
+         */
+        List<T> getResults();
+
+        /**
+         * @return The size of the result set.
+         */
+        Integer getSize();
+
+        /**
+         * @return The new offset from where the result set continues or 0 if the result set is finished.
+         */
+        Integer getNewOffset();
```

Comment:
I see, yeah that makes sense.

Reply:
So does this just return the same offset that was passed in the `setOffset` method?

I'm struggling a bit to understand the value in this.
I think the existing API was modeled on the JPA API, which is well understood. I think it would be a shame to diverge.

Reply:
As far as I understand, this will return the offset from which the result set continues.

For example, if you do a query with `offset=0`, `limit=100` then the `newOffset` will be 100. That means to get the next "page" of the result set you need to set your offset to `offset=100`. Obviously you can only do that if there's a next "page" (i.e. result set still has elements)

Reply:
But this is information you know, right? It's whichever offset you provided plus the limit.